File: Database/DTOs/PhotoDto.cs
using Database.DTOs.Subclasses;

namespace Database.DTOs;

public class PhotoDto : HasImage
{
    public string Id { get; set; } = string.Empty;

    /// <summary>Blob id in the photos container (for picking collection card image).</summary>
    public string ImageId { get; set; } = string.Empty;

    public required string FilmId { get; set; }

    public int Index { get; set; }

    /// <summary>1-based position in a collection when returned in collection context; null otherwise.</summary>
    public int? CollectionIndex { get; set; }

    public bool Restricted { get; set; }
}

File: analogagenda.client/src/app/DTOs/photo.dto.ts
import { HasImageDto } from ".";

export interface PhotoDto extends HasImageDto {
  id: string;
  /** Blob id in the photos container (for collection card image). */
  imageId?: string;
  filmId: string;
  index: number;
  /** 1-based order in a collection when applicable */
  collectionIndex?: number | null;
  restricted?: boolean;
}

export interface PhotoCreateDto {
  filmId: string;
  imageBase64: string;
  index?: number; // 0–999; client sets from upload order modal (API may assign if omitted)
}
